doc/guide/transactions-and-threading.rst

```diff
@@ -0,0 +1,394 @@
+============================
+Transactions and concurrency
+============================
+
+.. contents::
+
+`Transactions <https://en.wikipedia.org/wiki/Database_transaction>`_
+are a core feature of ZODB.  Much has been written about transactions,
+and we won't go into much detail here.  Transactions provide 2 core
+benefits:
+
+Atomicity
+  When a transaction executes, it succeeds or fails completely. If
+  some data are updated and then an error occurs, causing the
+  transaction to fail, the updates are rolled back automatically. The
+  application using the transactional system doesn't have to undo
+  partial changes.  This takes a significant burden from developers
+  and increases the reliability of applications.
+
+Concurrency
+  Transactions provide a way of managing concurrent updates to data.
+  Different programs operate on the data independently, without having
+  to use low-level techniques to moderate their access. Coordination
+  and synchronization happens via transactions.
+
```

Comment:
Seeing the *A* of ACID, I was expecting to see the *C*, but instead of *consistent* we get *concurrency*...and no *I* or *D*. Makes my head hurt :)

I know, it's simplified and the *D* is rolled into the *A* and the *I* into the *C*.

Pretty good simplification IMO.